network/client.go

```diff
@@ -151,8 +155,16 @@ func (c *PeerClient) Tell(message proto.Message) error {
 }
 
 // Request requests for a response for a request sent to a given peer.
-func (c *PeerClient) Request(req *rpc.Request) (proto.Message, error) {
-	signed, err := c.Network.PrepareMessage(req.Message)
+func (c *PeerClient) Request(ctx context.Context, req proto.Message) (proto.Message, error) {
```

Comment:
Maybe it will be better to add a new method (`RequestWithData` ?) to not a break backcompatibility and remove/rename it later. WDYT?

Reply:
Kenta said it was fine to break compatibility :)

Reply:
Let's go then ;D